### PR TITLE
GroupStore migration

### LINKED ISSUE #15523: Migrate AppError to error from store/sqlstore/group_store.go
Mattermost is migrating all usages of the `AppError` type to the more idiomatic `error` type. `AppError` is a loaded error type containing fields like HTTP Status code which is not really necessary for errors coming from the DB layer. We should migrate to using normal types satisfying the error interface.

This ticket is about migrating all usages of AppError from `store/sqlstore/group_store.go` to return a normal error. The migration should use the error types defined in `store/errors.go`. If the need should arise to add another specific error type, feel free to add it. For any generic error that is not something like an invalid input, or something not found, we can just wrap the error using `errors.Wrap(f)` and return it.

And then in the app layer, we would need to convert the `error` type to the usual `AppError`. Essentially moving the `AppError` creation from the store layer to the app layer. This means we need a switch case to check for the type of error and appropriately convert to the correct AppError. It can be something like:

```go
if _, err := a.Srv().Store.Bot().Update(bot); err != nil {
	var nfErr *store.ErrNotFound
	var appErr *model.AppError
	switch {
	case errors.As(err, &nfErr):
		return model.MakeBotNotFoundError(nfErr.Id)
	case errors.As(err, &appErr): // in case we haven't converted to plain error.
		return appErr
	default: // last fallback in case it doesn't map to an existing app error.
		return model.NewAppError("DeleteBotIconImage", "app.bot.patchbot.internal_error", nil, err.Error(), http.StatusInternalServerError)
	}
}
```

Also note that the error codes need to be changed from being a store level code (`store.<>.edit.error`) to something like `app.<>.edit.internal_error`.

Example Pull Request: #14339

If you have questions about the ticket or you need any help, feel free to contact `agniva.de.sarker` in https://community.mattermost.com/

----------------------------------

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-28812

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

Migration of `GroupStore` to return plain errors

#### Ticket Link

Fixes #15523

#### Jira ticket

https://mattermost.atlassian.net/browse/MM-28812

### CODE REVIEW

**Comment by rvillablanca** [comment]
This work was made on top of https://github.com/mattermost/mattermost-server/pull/15563 in order to reduce any merge conflict as this kind of refactor for an entire store are very prone to having lot of them. Once https://github.com/mattermost/mattermost-server/pull/15563 was merge I will ask for review for this PR.

**Comment by agnivade** [comment]
Over to you Rodrigo for the final piece. Thanks for all your work, and apologies for the delay on the user-store migration. I was on vacation past week.

**Comment by rvillablanca** [comment]
There are Elasticsearch failed tests. I assume it is not related with this PR unless my conflicts resolutions with master are wrong.

**Review by agnivade — COMMENTED** [review]
Nice job Rodrigo! Just one small typo.

File: app/group.go [hunk 1]
```diff
@@ -56,42 +126,88 @@ func (a *App) DeleteGroup(groupID string) (*model.Group, *model.AppError) {
...
+			return nil, model.NewAppError("UpsertGroupMember", "app.update_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+	}
+
+	return groupMember, nil
 }
 
 func (a *App) DeleteGroupMember(groupID string, userID string) (*model.GroupMember, *model.AppError) {
-	return a.Srv().Store.Group().DeleteMember(groupID, userID)
+	groupMemger, err := a.Srv().Store.Group().DeleteMember(groupID, userID)
```
[the remark this answers is not in the document]
Comment (rvillablanca) [reply app/group.go]: Fixed :+1:

**Review by agnivade — APPROVED** [review]

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @rvillablanca! :tada:

**Review by lindalumitchell — APPROVED** [review]

**Comment by agnivade** [comment]
Oh dear, I COMPLETELY forgot about this. So sorry about it. Going to send a PR today.

**Pull request merged**